### PR TITLE
operator: add DSA support

### PR DESCRIPTION
**Pull request by bart0sh**
Closes: #443

### CODE REVIEW

**Review by mythi — COMMENTED** [review]
thanks @bart0sh!

can you add `test/envtest` code too

File: pkg/controllers/dsa/controller.go [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+		},
+	}
+	// add the optional InitImage
+	if devicePlugin.Spec.InitImage != "" {
+		setInitContainer(&daemonSet.Spec.Template.Spec, devicePlugin.Spec.InitImage)
+	}
+	return &daemonSet
+}
+
+func setInitContainer(spec *v1.PodSpec, imageName string) {
```
[the remark this answers is not in the document]
Comment (bart0sh) [reply pkg/controllers/dsa/controller.go:180]: yep, I left it here in purpose. Some day we'll need to implement NFD support for this plugin. However, I'll be happy to remove it if you think it's better.

File: pkg/controllers/dsa/controller.go [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+								{
+									Name:      "devfs",
+									MountPath: "/dev/dri",
+									ReadOnly:  true,
+								},
+								{
+									Name:      "sysfs",
+									MountPath: "/sys/class/drm",
+									ReadOnly:  true,
+								},
```
[the remark this answers is not in the document]
Comment (bart0sh) [reply pkg/controllers/dsa/controller.go]: @mythi thank you for the review. Leftovers should be fixed now. Added envtest. Moved generated files into the separate commit.

File: pkg/controllers/dsa/controller.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mythi) [reply pkg/controllers/dsa/controller.go:193]: > Some day we'll need to implement NFD support for this plugin.

@bart0sh GPU/SGX plugins use this to install custom NFD feature sources but I cannot think of such need for DSA right now. Did you have something in mind?

[the remark this answers is not in the document]
Comment (bart0sh) [reply pkg/controllers/dsa/controller.go:193]: I thought that plugin would benefit from integration with NFD, but I'm not really familiar with it yet. If you think it doesn't make sense to do that I'll remove initcontainer-related code.

[the remark this answers is not in the document]
Comment (mythi) [reply pkg/controllers/dsa/controller.go:193]: @bart0sh I think we can leave it out in this version

[the remark this answers is not in the document]
Comment (bart0sh) [reply pkg/controllers/dsa/controller.go]: ok, removed

**Review by rojkov — APPROVED** [review]
Looks good to me!

**Review by mythi — APPROVED** [review]

**Pull request merged**